add support for hvf accelerator to qemu builder

adarobin opened the pull request:
This adds the newly released experimental 'hvf' accelerator as an option to the QEMU builder.  The accelerator adds support for Hypervisor.framework on macOS which significantly increases the performance of the QEMU builder on that platform.

Closes #6189

SwampDragons commented:
I don't have a qemu setup so I'm going to have to take your word for this working. I assume you've tested this?

SwampDragons commented on website/source/docs/builders/qemu.html.md.erb:
```diff
@@ -112,8 +112,8 @@ Linux server and have not enabled X11 forwarding (`ssh -X`).
-    This may be `none`, `kvm`, `tcg`, `hax`, or `xen`. The appropriate software
-    must have already been installed on your build machine to use the
+    This may be `none`, `kvm`, `tcg`, `hax`, `hvf`, or `xen`. The appropriate
+    software must have already been installed on your build machine to use the
```
Can you add a note that hvf is experimental? I'm always wary of just adding support for experimental stuff without hedging that there may be issues in the underlying tool.

adarobin commented:
@SwampDragons I have tested this with the only builds I am using QEMU with based on RHEL 6 and RHEL 7.  RHEL 6 worked with no changes (other than adjusting the accelerator).  RHEL 7 required adding ```[ "-global", "virtio-pci.disable-modern=on" ]``` to `qemuargs`.

I updated the documentation to highlight hvf is experimental and also added that qemu argument as possibly being needed.

SwampDragons commented:
Beautiful, thanks!
